pkg/login/social/generic_oauth.go

```diff
@@ -209,6 +213,43 @@ func (s *SocialGenericOAuth) extractFromToken(token *oauth2.Token) *UserInfoJson
 		return nil
 	}
 
+	var headerBytes []byte
+	headerBytes, err = base64.RawURLEncoding.DecodeString(matched[1])
+	if err != nil {
+		s.log.Error("Error base64 decoding header", "header", matched[1], "error", err)
+		return nil
+	}
+
+	var header map[string]string
+	err = json.Unmarshal(headerBytes, &header)
+	if compression, ok := header["zip"]; ok {
+		switch strings.ToLower(compression) {
+		case "gzip":
+			gr, err := gzip.NewReader(bytes.NewBuffer(rawJSON))
+			if err != nil {
+				s.log.Error("Error creating decompressor", "decoded_payload", string(rawJSON), "error", err)
+				return nil
+			}
+			defer gr.Close()
+			rawJSON, err = ioutil.ReadAll(gr)
+			if err != nil {
+				s.log.Error("Error decompressing payload", "decoded_payload", string(rawJSON), "error", err)
+				return nil
+			}
+		case "def":
+			fr := flate.NewReader(bytes.NewReader(rawJSON))
+			defer fr.Close()
+			rawJSON, err = ioutil.ReadAll(fr)
+			if err != nil {
+				s.log.Error("Error decompressing raw_payload", "decoded_payload", string(rawJSON), "error", err)
```

Comment:
Why is this different from :235?

```suggestion
				s.log.Error("Error decompressing payload", "decoded_payload", string(rawJSON), "error", err)
```

Reply:
copy/paste error, fixing